app/models/author.rb

```diff
@@ -121,44 +119,48 @@ def mirror_author_identities(import_settings)
     end
   end
 
+  # @param [Hash<String => [String, Hash]>] auth_hash
+  # @return [Hash<Symbol => String>]
   def self.build_attribute_hash_from_cap_profile(auth_hash)
-    # key/value not present in hash if value is not there
-    # sunetid/ university id/ ca licence ---- at least one will be there
+    # sunetid/ university id/ ca licence ---- at least one is expected
     seed_hash = {
       cap_profile_id: auth_hash['profileId'],
       active_in_cap:  auth_hash['active'],
       cap_import_enabled: auth_hash['importEnabled']
     }
-    profile = auth_hash['profile']
-    seed_hash[:sunetid] = profile['uid'] || ''
-    seed_hash[:university_id] = profile['universityId'] || ''
-    seed_hash[:california_physician_license] = profile['californiaPhysicianLicense'] || ''
-    seed_hash[:email] = profile['email'] || ''
-    seed_hash[:emails_for_harvest] = seed_hash[:email] # TODO: duplicate of :email
-    legal_name = profile['names']['legal']
-    pref_name = profile['names']['preferred']
-    seed_hash[:official_first_name] = legal_name['firstName'] || ''
-    seed_hash[:official_middle_name] = legal_name['middleName'] || ''
-    seed_hash[:official_last_name] = legal_name['lastName'] || ''
-    seed_hash[:cap_first_name] = pref_name['firstName'] || ''
-    seed_hash[:cap_middle_name] = pref_name['middleName'] || ''
-    seed_hash[:cap_last_name] = pref_name['lastName'] || ''
-    # TODO: preferred names are duplicates of :cap names
-    seed_hash[:preferred_first_name] = seed_hash[:cap_first_name]
+    profile    = Hash.new('').merge(auth_hash['profile'])
+    legal_name = Hash.new('').merge(profile['names']['legal'])
+    pref_name  = Hash.new('').merge(profile['names']['preferred'])
+    seed_hash[:sunetid] = profile['uid']
+    seed_hash[:university_id] = profile['universityId']
+    seed_hash[:california_physician_license] = profile['californiaPhysicianLicense']
+    seed_hash[:email] = profile['email']
+    seed_hash[:official_first_name]  = legal_name['firstName']
+    seed_hash[:official_middle_name] = legal_name['middleName']
+    seed_hash[:official_last_name]   = legal_name['lastName']
+    seed_hash[:cap_first_name]  = pref_name['firstName']
+    seed_hash[:cap_middle_name] = pref_name['middleName']
+    seed_hash[:cap_last_name]   = pref_name['lastName']
+    # TODO: duplicative values
+    seed_hash[:emails_for_harvest]    = seed_hash[:email]
+    seed_hash[:preferred_first_name]  = seed_hash[:cap_first_name]
     seed_hash[:preferred_middle_name] = seed_hash[:cap_middle_name]
-    seed_hash[:preferred_last_name] = seed_hash[:cap_last_name]
+    seed_hash[:preferred_last_name]   = seed_hash[:cap_last_name]
     seed_hash
```

Comment:
All these changes to `seed_hash` are puzzling and it's difficult to see exactly what is just style/whitespace and what is functional changes (if any).  The previous code seems to default to an empty string `''` but the new code will default to `nil` values.  That seems like a functional change and could have some impact.  What are the consequences of this change?

Reply:
Also puzzled by the idea of the hash default, consider:
```ruby
[15] pry(main)> x = Hash.new('')
=> {}
[16] pry(main)> x[:abc]
=> ""
[17] pry(main)> x[:abc] = nil
=> nil
[18] pry(main)> x[:abc]
=> nil
```

Reply:
It's the repetition of a dozen of failovers to the same default value that calls for a Hash default, e.g.:

```ruby
seed_hash[:cap_first_name] = pref_name['firstName'] || ''
seed_hash[:cap_middle_name] = pref_name['middleName'] || ''
seed_hash[:cap_last_name] = pref_name['lastName'] || ''
...
```

I agree w/ doing more in a single block, where possible (I think I changed the initial lines to at least do the first 3 at once) and you're right that this gets us closer to that.  The exceptions are the `duplicative values` which rely on values already being set (or having to repeat dives down levels of incoming hash).

Reply:
You are right about the behavior where an explicit `nil` value is assigned.  The default is invoked when the hash has no matching key.  That's why I documented the expectations about the incoming parameter as:
```ruby
@param [Hash<String => [String, Hash]>]
```
I.E., `nil` values are not expected.  (Of course this is not comprehensive since I didn't fully specify the nested Hash characteristics, but you get the idea.)   If this concern is material, we should make a test case for it.  The old approach could be applied or filters on the instantiation of the smaller hashes.

Reply:
Perhaps another syntax option is a fetch, e.g. `pref_name.fetch('firstName', '')`.  In general, it's a pity the key strings are all mapped to different strings.

Reply:
Another option for nested hash, where intermediates might be nil, is `dig` (but it doesn't have a default option like fetch):
- https://docs.ruby-lang.org/en/2.4.0/Hash.html#method-i-dig